### DISCUSSION TITLE
Context in TMX

### DISCUSSION THREAD

**Discussion by shrihari-k** [WeblateOrg/weblate#9930] [Q&A]
Below is a sample PO file I used. But the msgctxt is not there in the TMX when I download? What I am missing to get this in to the TMX so that next time it will be more contextual when the TMX selection happens

msgid "Friend"
msgstr "Ami(e)"

msgctxt "FEMALE"
msgid "Friend"
msgstr "Amie"

msgctxt "MALE"
msgid "Friend"
msgstr "Ami"

**Comment by nijel** [comment]
I don't think TMX supports context. At least there is no implementation of that in https://github.com/translate/translate/blob/master/translate/storage/tmx.py.

**Reply by shrihari-k** [reply]
Thank you, Michel. But then when we upload the PO files we lose all the context information that we have captured for future use. Will it not be appropriate to have this context information available in the TMX so that the translation matching will be more accurate?

**Reply by nijel** [reply]
Yes, it would be useful to some extent, but I don't currently see TMX format supporting this.

Still, using translation memory is like using machine translation – you can never 100% rely on it to provide accurate results.

**Comment by nijel** [comment]
There is now also an issue for this: https://github.com/WeblateOrg/weblate/issues/9947

**Reply by shrihari-k** [reply]
Agree. The following are features & usecases I am thinking of.
1. TMX data should have an option to be used with context / without context during translation
2. To have an option for TMX to be download with context / without context
Following are the use cases I am envisioning 
1. When there is an incremental update we know that contextually things have not changed then we can look at an option to possibly avoid manual review and save time.
2. To have the TMX updated with context so that the information in the PO / XLIFF file when imported is not lost for future use if someone wants to use it (since PO is a temporary file). We will have the information in the repo.
